daprdocs/content/en/developing-applications/building-blocks/state-management/howto-get-save-state.md

```diff
@@ -656,6 +656,53 @@ dapr run --app-id orderprocessing --app-port 6001 --dapr-http-port 3601 --dapr-g
 
 {{% codetab %}}
 
+```go
+// dependencies
+import (
+	"context"
+	"log"
+	"math/rand"
+	"strconv"
+	"time"
+
+	dapr "github.com/dapr/go-sdk/client"
+)
+
+// code
+func main() {
+	const STATE_STORE_NAME = "statestore"
+	rand.Seed(time.Now().UnixMicro())
+	for i := 0; i < 10; i++ {
+		orderId := rand.Intn(1000-1) + 1
+		client, err := dapr.NewClient()
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+		ctx := context.Background()
+		err = client.SaveState(ctx, STATE_STORE_NAME, "order_1", []byte(strconv.Itoa(orderId)), nil)
+		if err != nil {
+			panic(err)
+		}
+		keys := []string{"key1", "key2", "key3"}
+        items, err := client.GetBulkState(ctx, store, keys, nil, 100)
+
+		log.Println("Result after get:", string(result.Value))
```

Comment:
This will not work, as this is an array of BulkStateItems, what's the objective of this println? show all the Items in the array? We might need to loop  here.

Reply:
@salaboy
Yes, @hhunter-ms  is asking the the equivalent of this below, but for bulkstate, which means iterating over each item and showing it's value. This is just for a simple example to show the API works

```
		result, err := client.GetState(ctx, STATE_STORE_NAME, "order_1", nil)
		if err != nil {
			panic(err)
		}
		log.Println("Result after get:", string(result.Value))
```

Reply:
Ok, then @hhunter-ms something like this would work
```
                 items, err := client.GetBulkState(ctx, STATE_STORE_NAME, keys, nil, 100)
		if err != nil {
			panic(err)
		}
		for _, item := range items {
			log.Println("Item from GetBulkState:", string(item.Value))
		}
```

Reply:
Thanks @salaboy! I've pushed that update